Models/UserModel.php

```diff
@@ -1,100 +1,218 @@
 <?php
+declare(strict_types=1);
 
 namespace Models;
 
+/**
+ * @package Models
+ */
 class UserModel extends DataModel
 {
-    public function login($login, $password)
+    /**
+     * Check type of user and login on site
+     *
+     * @param string $login
+     * @param string $password
+     * @return int
+     */
+    public function login(string $login, string $password): int
     {
         $user = $this->userExist($login);
-        if ($user && password_verify($password, $user[0]['password']) && ($user[0]['approve_status'])) {
-            $_SESSION['login'] = $login;
-            $_SESSION['id'] = $user[0]['id'];
-            return 1;
-        } elseif ($user && !$user[0]['approve_status']) {
+        if ($user && password_verify($password, $user[0]['password'])) {
+            if (!isset($user[0]['approve_status']) || $user[0]['approve_status']) {
+                $this->setSessionData($user[0]);
+                if ($this->sessionData->getUser()['type'] == 'students') {
+                    return 3;
+                }
+                return 1;
+            }
             return 2;
         }
         return 0;
```

Comment:
please make a variable for this result and return variable once instead of declaring return statement four times inside one function